### PR TITLE
Enhance gen csr API

### PR DESCRIPTION
**Pull request by Wenxing-hou**
Fix: #2002 

Commit1:
1.add opaque data input;
2.change Change uint8_t **csr_pointer to uint8_t *csr_pointer;
3.delete LIBSPDM_MAX_CSR_SIZE;

Commit2: 
Fix unit_test after enhance gen_csr api;

### CODE REVIEW

File: include/hal/library/cryptlib/cryptlib_cert.h [hunk 1]
```diff
@@ -412,7 +412,7 @@ extern bool libspdm_sm2_get_public_key_from_x509(const uint8_t *cert, size_t cer
 extern bool libspdm_gen_x509_csr(size_t hash_nid, size_t asym_nid,
                                  uint8_t *requester_info, size_t requester_info_length,
                                  void *context, char *subject_name,
-                                 size_t *csr_len, uint8_t **csr_pointer);
+                                 size_t *csr_len, uint8_t *csr_pointer);
```
Comment (steven-bellock) [inline include/hal/library/cryptlib/cryptlib_cert.h:415]: I just noticed that `libspdm_gen_x509_csr` is not used by any of the core libraries. Please move this declaration to https://github.com/DMTF/libspdm/blob/main/os_stub/spdm_crypt_ext_lib/cryptlib_ext.h

**Review by steven-bellock — CHANGES_REQUESTED** [review]

File: os_stub/cryptlib_mbedtls/pk/x509.c [hunk 2]
```diff
@@ -1834,7 +1835,13 @@ bool libspdm_gen_x509_csr(size_t hash_nid, size_t asym_nid,
 
     *csr_len = (size_t)ret;
     /*change csr_pointer to start location*/
-    *csr_pointer = *csr_pointer + csr_buffer_size - *csr_len;
+
+    index = 0;
+
+    while(index < *csr_len) {
```
[the remark this answers is not in the document]
Comment (Wenxing-hou) [reply os_stub/cryptlib_mbedtls/pk/x509.c]: Thanks. I have changed the code.

File: unit_test/fuzzing/test_requester/test_spdm_requester_get_csr/get_csr.c [hunk 3]
```diff
@@ -179,10 +179,10 @@ void libspdm_test_requester_get_csr_case1(void **State)
 {
     libspdm_test_context_t *spdm_test_context;
     libspdm_context_t *spdm_context;
-    uint8_t csr_form_get[LIBSPDM_MAX_CSR_SIZE] = {0};
+    uint8_t csr_form_get[0x1000] = {0};
```
Comment (steven-bellock) [inline unit_test/fuzzing/test_requester/test_spdm_requester_get_csr/get_csr.c:182]: You can define `LIBSPDM_MAX_CSR_SIZE` for unit tests / os_stub.

Comment (Wenxing-hou) [reply unit_test/fuzzing/test_requester/test_spdm_requester_get_csr/get_csr.c]: Thanks. I have defined the `LIBSPDM_MAX_CSR_SIZE` for unit tests.

**Pull request merged**